### ISSUE TITLE
Make http::client::make_request() abortable

### ISSUE THREAD

**Issue by xemul** [scylladb/seastar#2409]
Currently the method can block in one of 4 places:
- connecting to server
- waiting for a connection from pool
- writing request to socket
- reading response from server

Either of the steps should be abortable. One of the use-cases is to time-out the request (see #2304). Another -- abort of large file upload in case this upload by user request (e.g. cancelling backup process for whatever reason)

**Comment by ndragazis** [comment]
@xemul the first bullet was never fully implemented, right? In https://github.com/scylladb/seastar/commit/35badf56f1d145de520c6e1aa3f4526492026b88 you mention that it will be done as a followup. Do we have an issue about this?

**Comment by xemul** [comment]
@ndragazis , yes, connect() still blocks. We don't have special issue for it, but we have known issue _with_ it: https://github.com/scylladb/seastar/issues/2302
